### PR TITLE
CI: update to self-hosted runner

### PR DESCRIPTION
**Pull request by mateuszskrobek**
## Purpose

The purpose it to test CI on self-hosted runner that is using docker containers.

## Proposed Changes

Modifies CI workflows files.

## Issues

N/A

## Testing

This draft PR is test itself as only this way we can test CI jobs execution.

### CODE REVIEW

File: .github/workflows/poetry-test.yml [hunk 1]
```diff
@@ -9,19 +9,30 @@ on:
...
       image: osrf/ros:${{ matrix.ros_distro }}-desktop-full
 
     steps:
+      - name: Clean up workspace
+        run: |
+          ls -la $GITHUB_WORKSPACE
+          rm -rf $GITHUB_WORKSPACE/* || true
+          rm -rf $GITHUB_WORKSPACE/.??* || true
+          ls -la $GITHUB_WORKSPACE
+
```
Comment (maciejmajek) [inline .github/workflows/poetry-test.yml:35]: Is this step needed?

File: .github/workflows/pre-commit.yml
Comment (maciejmajek) [inline .github/workflows/pre-commit.yml:1]: Could you please revert the additional \n?

File: .github/workflows/licensechecker.yaml
Comment (maciejmajek) [inline .github/workflows/licensechecker.yaml:1]: Could you please revert the additional \n s?

File: .github/workflows/poetry-test.yml (hunk 1 above)
Comment (mateuszskrobek) [reply .github/workflows/poetry-test.yml:35]: I think so, as it cleans the repo with build, install & log directories (check workflows logs for the reference). `ls -la $GITHUB_WORKSPACE` can be removed as it only lists files before and after the clean-up however, it makes it easier to debug and validate if `rm -rf` command actually worked.

File: .github/workflows/pre-commit.yml
Comment (mateuszskrobek) [reply .github/workflows/pre-commit.yml:1]: Done

File: .github/workflows/licensechecker.yaml
Comment (mateuszskrobek) [reply .github/workflows/licensechecker.yaml:1]: Done

**Review by maciejmajek — APPROVED** [review]
Thank you, LGTM

**Pull request merged**